app/angular/src/server/config/webpack.config.js

```diff
@@ -55,6 +57,16 @@ export default function() {
       new CaseSensitivePathsPlugin(),
       new WatchMissingNodeModulesPlugin(nodeModulesPaths),
       new webpack.ProgressPlugin(),
+      // temp plugin to make webpack bundle only one v5 version.
```

Comment:
when changing a version of the `example/angular-cli` to different than the one in the `app/angular` or `addons/knbos` -  the v5 is bundled twice and causes to `StaticInjectorError something-something`. I wonder will it be the same if storybook is used in a separate project (I think yes), should we put this plugin to the code base or document that people will need to add it explicitly

Reply:
Please correct me if I'm wrong (probably yes) but shouldn't the angular version inside the `@storybook/angular`s `package.json` be used as a dep of storybook and the version in the consumer project be used for that (when using storybook in a separate project)?

Anyway, we'll need to test it.

Reply:
Let's say there is a project with this package.json

```
{
  "dependencies": [
     "@angular/core": "^4.x.x"
  ],
  "devDependencies": [
     "@storybook/angular": "^3.3"
     "@storybook/addon-knobs": "^3.3"
  ]
}
```

I assume that the node modules structure will be like this:
```
node_modules
  |- angular
  |-- core (4.x.x)
  |- storybook
  |-- angular
  |--- node_modules
  |---- angular
  |----- core (5.x.x)
  |-- addon-knobs
  |--- node_modules
  |---- angular
  |----- core (5.x.x)
```
So what webpack will do...

Reply:
uhm, shouldn't `@angular/core` be a peerDependency to prevent exactly that?

Reply:
@ndelangen, yeah, that was my first implementation. But after, we discussed it in [this](https://storybooks.slack.com/archives/C655M4PGR/p1513091235000753) long conversation. In short, having a peer on angular will force us to be super backward compatible with the old versions, and we won't be able to adopt new angular things. For example, there is a thingy called `InjectionToken` that used in `app/angular`. It was introduced in v4 instead of the old `OpaqueToken` from v2 (that became deprecated in v4). In v5 the `OpaqueToken` was completely removed. In this case, with a peer on angular, we can't support the v2 (or we need to add some webpack replacement hooks). The alternative is to depend on the latest angular version, but in this case, we end up with 2 angular versions in the bundle (or 3 because of the described above). I hope I was clear 😞

Reply:
Well to be honest, I think the situation of "it's a peerDependency and you have to make sure it's a version that's compatible with storybook/angular" is preferable to "it's a regularDependency and you better be using the exact same version or you're f%*&ed".

Reply:
So if returning it back to peer we will have to deprecate angular 2. I am ok with this if you ask me, but I'm not familiar with the adoption rate of angular..

Reply:
Is there any way to do some kind of feature detection and pick between `OpaqueToken` and `InjectionToken` based on what's available in actual angular version?

Reply:
That what I've tried to do with webpack replacement, but turned out of this way. I think it won't scale if will need to patch every breaking change...